client/client.go

```diff
@@ -185,9 +199,27 @@ func NewClientWithContext(ctx context.Context, pdAddrs []string, security Securi
 
 func (c *client) updateURLs(members []*pdpb.Member) {
 	urls := make([]string, 0, len(members))
+	urlsMap := make(map[string]struct{})
 	for _, m := range members {
 		urls = append(urls, m.GetClientUrls()...)
 	}
+	for _, url := range urls {
+		urlsMap[url] = struct{}{}
+	}
+	containsOldURLs := true
```

Comment:
how about `hasNewURL := false` and Line212 to `hasNewURL = true`

Reply:
maybe clearer to simply sort 2 `[]string` then `reflect.DeepEqual`.

Reply:
I like it.